### PR TITLE
Re-enable golangci-lint

### PR DESCRIPTION
**Pull request by rafaelvanoni**

### CODE REVIEW

File: internal/pkg/testutils/utils.go [hunk 1]
```diff
@@ -269,8 +269,6 @@ func CreateContainer(netconf, podName, podNamespace, ip string) (containerID str
 }
 
 // Create container with the giving containerId when containerId is not empty
-//
-// Deprecated: Please call CreateContainerNamespace and then RunCNIPluginWithID directly.
```
Comment (caseydavenport) [inline internal/pkg/testutils/utils.go]: Why did you remove this comment? I think we don't want people using the function in the test code.

File: internal/pkg/testutils/utils.go [hunk 2]
```diff
@@ -584,12 +581,11 @@ func AddNode(c client.Interface, kc *kubernetes.Clientset, host string) error {
 }
 
 func DeleteNode(c client.Interface, kc *kubernetes.Clientset, host string) error {
-	var err error
-	err = nil
+	var err error = nil
```
Comment (caseydavenport) [inline internal/pkg/testutils/utils.go]: I don't see us ever using this - shouldn't we just change line 596 to `return nil`?

File: internal/pkg/testutils/utils.go (hunk 1 above)
Comment (rafaelvanoni) [reply internal/pkg/testutils/utils.go]: One of the linters enforces `Deprecated` comments, but I spoke to @ozdanborne and he told me to ignore it and stick with the current call.

File: internal/pkg/testutils/utils.go (hunk 2 above)
Comment (rafaelvanoni) [reply internal/pkg/testutils/utils.go]: We still use `err` on line 593 and return that value.

**Comment by caseydavenport** [comment]
@rafaelvanoni LGTM - could you squash before merging?
